src/pages/ServiceDetails/ServiceDetailsPage.tsx

```diff
@@ -322,18 +323,29 @@ class ServiceDetails extends React.Component<ServiceDetailsProps, ServiceDetails
           }
         });
     } else {
-      const params = getQueryJaeger();
+      let params = changeParams(this.traceParams);
```

Comment:
I had in mind something simpler with just a defensive copy of the params:

```javascript
const params = getQueryJaeger();
const fetchParams = { ...params }; // copy
if (this.lastFetchTraceMicros) {
  fetchParams[URLParam.JAEGER_START_TIME] = this.lastFetchTraceMicros;
}
fetchTraces(this.props.match.params.namespace, this.props.match.params.service, fetchParams)
```
This way when you modify fetchParam it doesn't affect the "global" params object.

Reply:
yes but I can't check if the user change any option with this like tags or limit